static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -57,50 +137,102 @@ function WidgetBuilder({
     query: {...location.query, dataSet: undefined},
   };
 
-  const [title, setTitle] = useState(
-    t('Custom %s Widget', displayTypes[DisplayType.AREA])
-  );
+  // Construct PageFilters object using statsPeriod/start/end props so we can
+  // render widget graph using saved timeframe from Saved/Prebuilt Query
+  const pageFilters: PageFilters = statsPeriod
+    ? {...selection, datetime: {start: null, end: null, period: statsPeriod, utc: null}}
+    : start && end
+    ? {...selection, datetime: {start, end, period: null, utc: null}}
+    : selection;
+
+  // when opening from discover or issues page, the user selects the dashboard in the widget UI
+  const omitDashboardProp = [
```

Comment:
I'm wondering if we need this in the full page builder. I checked out the `addDashboardWidgetModal.tsx` file and this is used to trigger a fetch for the dashboards. If we're already on a dashboard that we're going to add to, I don't think this is necessary

Reply:
I think this is because we can also open the widget builder modal in"discover" and in other verticals of our application, but let's see...maybe we can update it in the future